ext/bg/js/backend.js

```diff
@@ -809,6 +772,20 @@ class Backend {
         return portName;
     }
 
+    async _onApiModifySettings({targets, source}) {
+        const results = [];
+        for (const target of targets) {
+            try {
+                this._modifySetting(target);
+                results.push({result: true});
+            } catch (e) {
+                results.push({error: errorToJson(e)});
+            }
+        }
+        await this._onApiOptionsSave({source});
```

Comment:
Is the `source` argument actually used anywhere?

Reply:
It's used on the settings page to determine what the source was for a settings change. It uses this value to not get into infinite loops of updating values due to a change that was triggered by itself.

Would be good to get rid of if possible, but the current setup made that difficult at the time.